XiEditor/Search.swift

```diff
@@ -196,19 +232,55 @@ extension EditViewController {
             findViewController.ignoreCase = status.first?["case_sensitive"] as! Bool
         }
 
+        if status.first?["whole_words"] != nil && !(status.first?["whole_words"] is NSNull) {
+            findViewController.wholeWords = status.first?["whole_words"] as! Bool
+        }
+
         if let resultCount = status.first?["matches"] as? Int {
             (findViewController.searchField as? FindSearchField)?.resultCount = resultCount
         }
     }
-    
+
+    func replaceNext() {
+        document.sendRpcAsync("replace_next", params: [])
+    }
+
+    func replaceAll() {
+        document.sendRpcAsync("replace_all", params: [])
+    }
+
+    func replaceStatus(status: [String: AnyObject]) {
+        if status["chars"] != nil && !(status["chars"] is NSNull) {
+            findViewController.replaceField.stringValue = status["chars"] as! String
+        }
+
+        // todo: preserve case
+    }
+
+    func replace(_ term: String?) {
+        var params: [String: Any] = [
+            "preserve_case": false,     // todo: impement option for preserving case
```

Comment:
Are you thinking you'll implement this in the near future? In general I'm reluctant to add unused params, but if it's next up then no problem. 👍

Reply:
Preserving case is quite hard. I wasn't planning on working on it next, it would be future work. So, I guess I will remove it for now.